pkgs/servers/aeron/default.nix

```diff
@@ -0,0 +1,168 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  jdk11,
+  gradle,
+  makeWrapper,
+  glib,
+  perl,
+  writeText
+}:
+
+let
+  pname = "aeron";
+  version = "1.40.0";
+
+  src = fetchFromGitHub {
+    owner = "real-logic";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-4C5YofA/wxwa7bfc6IqsDrw8CLQWKoVBCIe8Ec7ifAg=";
+  };
+
+  deps = stdenv.mkDerivation {
+    name = "${pname}-deps";
+    inherit src;
+
+    nativeBuildInputs = [
+      gradle
+      jdk11
+      perl
+    ];
+
+    buildPhase = ''
+      export GRADLE_USER_HOME=$(mktemp -d);
+      gradle \
+          --no-daemon \
+        build \
+          -Dorg.gradle.java.home="${jdk11.home}" \
+          -x test \
+          -x checkstyleMain \
+          -x checkstyleGenerated \
+          -x checkstyleGeneratedTest \
+          -x checkstyleMain \
+          -x checkstyleTest
+    '';
+
+    # Mavenize dependency paths
+    # e.g. org.codehaus.groovy/groovy/2.4.0/{hash}/groovy-2.4.0.jar -> org/codehaus/groovy/groovy/2.4.0/groovy-2.4.0.jar
+    installPhase = ''
+      find "$GRADLE_USER_HOME/caches/modules-2" -type f -regex '.*\.\(jar\|pom\)' \
+        | perl -pe 's#(.*/([^/]+)/([^/]+)/([^/]+)/[0-9a-f]{30,40}/([^/\s]+))$# ($x = $2) =~ tr|\.|/|; "install -Dm444 $1 \$out/$x/$3/$4/$5" #e' \
+        | sh
+      ln -s "$out/com/squareup/okio/okio/2.10.0/okio-jvm-2.10.0.jar" "$out/com/squareup/okio/okio/2.10.0/okio-2.10.0.jar"
+    '';
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = "sha256-QRL3y7HF6/pWsC5MbtIqhZ4HR8jcWtQG6x9kajeoNMY=";
+  };
+
+  # Point to our local deps repo
+  gradleInit = writeText "init.gradle" ''
+    settingsEvaluated { settings ->
+      settings.pluginManagement {
+        repositories {
+          clear()
+          maven { url '${deps}' }
+        }
+      }
+    }
+    gradle.projectsLoaded {
+      rootProject.allprojects {
+        repositories {
+          clear()
+          maven { url '${deps}' }
+        }
+      }
+    }
+  '';
+
+  # replace buildSrc
+  buildSrc = writeText "build.gradle" ''
+    repositories {
+      clear()
+      maven { url '${deps}' }
+    }
+
+    dependencies {
+      implementation 'org.asciidoctor:asciidoctorj:2.5.5'
+      implementation 'org.eclipse.jgit:org.eclipse.jgit:5.13.1.202206130422-r'
+    }
+  '';
+
+in stdenv.mkDerivation rec {
+
+  inherit pname src version;
+
+  nativeBuildInputs = [
+    glib
+    gradle
+    jdk11
+    makeWrapper
```

Comment:
this does not look cross ready. glib probably goes to buildInputs and jdk11, too unless it is run at build time

Reply:
glib was included in error, removing.

I'm not sure of the best practice for including jdk11 - do runtime paths normally refer to pkgs.jdk11, or pkgs.jre?